File: Sort/BubbleSort.js
/*
 * Problem: Bubble sort
 * Author: Dung Nguyen Tien (Chris)
*/
function bubbleSort(arr) {
  var noSwap = true;
  for (var len = arr.length; len > 0; len--) {
    noSwap = true
    for (var i = 0; i < len - 1; i++) {
      if (arr[i] > arr[i + 1]) {
        [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]];
        noSwap = false;
      }
    }
    // if the array is sorted already
    if (noSwap) break;
  }
}

var arr = [1,5,3,7,8,3, 0];
bubbleSort(arr);
document.writeln(JSON.stringify(arr));

File: Sort/BubbleSort.cs
/*
 * Problem: BubbleSort
 * Author: Dung Nguyen Tien (Chris)
*/

public static void BubbleSort(List<int> arr)
{
	var noSwap = true;
	for (var length = arr.Count; length > 0; length--)
	{
		noSwap = true;
		for (var i = 0; i < length - 1; i++)
		{
			if (arr[i] > arr[i + 1])
			{
				noSwap = false;
				Swap(arr, i, i + 1);
			}
		}

		if (noSwap) break;
	}
}

public static void Swap(List<int> arr, int x, int y)
{
	if (x == y) return;

	var tmp = arr[x];
	arr[x] = arr[y];
	arr[y] = tmp;
}